modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java

```diff
@@ -367,39 +347,50 @@ private synchronized void updateAccountState(Address address, AccountState accou
             if (accountState != null) {
                 storageRoot = getAccountState(address).getStateRoot();
             }
-            IContractDetails<DataWord> details = detailsDS.get(address.toBytes());
+
+            details = detailsDS.get(address.toBytes());
 
             if (details != null) {
                 details = details.getSnapshotTo(storageRoot);
             }
-
-            return details;
         } finally {
             rwLock.readLock().unlock();
+            return details;
```

Comment:
May lead to unintended errors later, since this ``details`` will always be the return regardless of whether we add another ``return`` inside the try